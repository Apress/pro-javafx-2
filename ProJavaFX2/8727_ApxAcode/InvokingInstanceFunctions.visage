/*
 * @test
 * @run
 */

import java.lang.Math.*;

class Point {
  var x: Number;
  var y: Number;
  function distanceFromOrigin(): Number {
    sqrt(x*x + y*y)
  }
  function translate(dx: Number, dy: Number) {
    x += dx;
    y += dy;
  }
  override function toString(): String {
    "Point({x}, {y})"
  }
}

var p = Point { x: 3.0, y: 4.0 };
println("p = {p}");
println("Distance between p and the origin = {p.distanceFromOrigin()}");

p.translate(2.0, 8.0);
println("p = {p}");
println("Distance between p and the origin = {p.distanceFromOrigin()}");

print("Distance between Point \{x: 8.0, y: 15.0\} and the origin = ");
println(Point {x: 8.0, y: 15.0}.distanceFromOrigin());
