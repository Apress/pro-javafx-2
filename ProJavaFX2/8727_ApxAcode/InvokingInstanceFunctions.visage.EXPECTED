p = Point(3.0, 4.0)
Distance between p and the origin = 5.0
p = Point(5.0, 12.0)
Distance between p and the origin = 13.0
Distance between Point {x: 8.0, y: 15.0} and the origin = 17.0
